src/ui/render.ts

```diff
@@ -31,15 +35,39 @@ export default class UI {
       selectionStart: null,
       selectionEnd: null,
     };
+    this.latestSelectedCell = undefined;
     this.registerEvents();
     this.isReadOnly = lightSheet.options.isReadOnly || false;
 
     this.tableEl.classList.add("lightsheet_table_container");
 
+    /*content*/
     const lightSheetContainerDom = document.createElement("div");
     lightSheetContainerDom.classList.add("lightsheet_table_content");
     this.tableEl.appendChild(lightSheetContainerDom);
 
+    /*formula bar*/
+    this.lightSheetFormulaBarDom = document.createElement("div");
+    this.lightSheetFormulaBarDom.classList.add("light_sheet_table_formula_bar");
+    lightSheetContainerDom.appendChild(this.lightSheetFormulaBarDom);
+
+    //selected cell display element
+    this.selectedCellDisplay = document.createElement("div");
+    this.selectedCellDisplay.classList.add("selected_cell_display");
+    this.lightSheetFormulaBarDom.appendChild(this.selectedCellDisplay);
+
+    //"fx" label element
+    const fxLabel = document.createElement("div");
+    fxLabel.textContent = "fx";
+    fxLabel.classList.add("fx_label");
+    this.lightSheetFormulaBarDom.appendChild(fxLabel);
+
+    //formula input
+    this.lightSheetFormulaInput = document.createElement("input");
+    this.lightSheetFormulaInput.classList.add("formula_input");
+    this.lightSheetFormulaBarDom.appendChild(this.lightSheetFormulaInput);
+    this.setFormulaBar();
```

Comment:
The constructor is getting too large, seperating them into methods would be nice